### PR TITLE
Add warning about array casting

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

File: src/swarm/neo/util/AcquiredResources.d [hunk 1]
```diff
@@ -66,6 +66,26 @@ public struct AcquiredArraysOf ( T )
...
+            struct S { int i; hash_t h; }
+            auto s_array = cast(S[]*)void_array;
+            s_array.length = 23;
+        ---
+
+        The exact reason for this is not known, but it appears to lead to
+        corruption of internal GC data (possibly type metadata associated with
+        the array's pointer). Setting the length of the array *before* casting
+        is safe.
+
```
Comment (david-eckardt-sociomantic) [inline src/swarm/neo/util/AcquiredResources.d]: Suggest adding a positive example, such as
```d
(*void_array).length = 23 * S.sizeof; // First set the buffer length...
auto s_array = cast(S[])*void_array;   // ... then cast the array.
```
Maybe even change the order: First say how one should do it, then mention a way that is known to fail.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/AcquiredResources.d]: Yep, good idea.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/AcquiredResources.d]: Updated.

**Comment by gavin-norman-sociomantic** [comment]
Rebased.

**Review by david-eckardt-sociomantic — APPROVED** [review]

**Pull request merged**